measurements/pages/__init__.py

```diff
@@ -275,6 +275,18 @@ def generator():
 
 @pages_blueprint.route('/files/download/<path:textname>')
 def files_download(textname):
+    if len(textname.split('/')) == 1:
+        # This is for backward compatibility with the new pipeline.
+        # See: https://github.com/TheTorProject/ooni-measurements/issues/44
+        q = current_app.db_session.query(Report.textname)\
+                .filter(Report.textname.endswith(textname))
```

Comment:
Does `/files/download/.json` lead to slow query following an OOM?
Should it be `'/' + textname`?
Should `.limit(2)` be added?
Should exception be raised on two rows returned? IIRC, there were some reports that that same name in two buckets. Probably, these reports are duplicated, but I don't remember by heart.

Reply:
It does lead to pretty slow queries, but they are not that bad (on very slow network it's about 5-6 second)

Reply:
How much RAM, CPU and network transfer does suggested pathological case take?
What query does it actually generate? Does `q.first()` implicitly add `LIMIT 1` or not?

Reply:
> first() applies a limit of one within the generated SQL, so that only one primary entity row is generated on the server side
http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.first